screens/Home/HomeScreen.js

```diff
@@ -4,61 +4,72 @@ import {
   View,
   Image,
   ScrollView,
-  Pressable,
   Text,
   TouchableOpacity,
+  ActivityIndicator,
 } from "react-native";
 import SearchComp from "../../components/HomeScreenComp/SearchComp";
 import CardComp from "../../components/HomeScreenComp/CardComp1";
 import CardComp2 from "../../components/HomeScreenComp/CardComp2";
 import Fontisto from "react-native-vector-icons/Fontisto";
 import { useNavigation } from "@react-navigation/native";
 import RBSheet from "react-native-raw-bottom-sheet";
-import ContactPermission from "../../components/HomeScreenComp/contactPermission";
-import NotificationPermission from "../../components/HomeScreenComp/notificationPermission";
-import NewNotify from "../NewNotification";
 import * as Contacts from "expo-contacts";
-import * as Notifications from "expo-notifications";
+import axios from "axios";
 
 import SlideAlert from "../../components/TostMessage/SlideAlert";
 import PopularCards from "../../components/HomeScreenComp/PopularCard/PopularCards";
 import { AuthContext } from "../../store/auth-context";
 
 export default function HomeScreen() {
+  const [apiData, setapiData] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
   const refRBSheet = useRef();
   const navigation = useNavigation();
 
-  const [notificationCount, setNotificationCount] = useState(5); // Set the actual count as needed
-  const [notificationPermission, setNotificationPermission] = useState(false);
+  const context = useContext(AuthContext);
+  console.log(context.token);
 
   useEffect(() => {
-    const checkAndRequestContactPermission = async () => {
-      const { status } = await Contacts.getPermissionsAsync();
+    const fetchData = async () => {
```

Comment:
This Data fetching  , API calling and etc. should be done in respective API files. eg. for this HomeScreen page we HomeAPI.js file in api folder.